spring-cloud-circuitbreaker-resilience4j/src/main/java/org/springframework/cloud/circuitbreaker/resilience4j/Resilience4JAutoConfiguration.java

```diff
@@ -52,13 +57,33 @@
 	@Bean
 	@ConditionalOnMissingBean(CircuitBreakerFactory.class)
 	public Resilience4JCircuitBreakerFactory resilience4jCircuitBreakerFactory(
-			CircuitBreakerRegistry circuitBreakerRegistry, TimeLimiterRegistry timeLimiterRegistry) {
+			CircuitBreakerRegistry circuitBreakerRegistry, TimeLimiterRegistry timeLimiterRegistry,
+			@Autowired(required = false) Resilience4jBulkheadProvider bulkheadProvider) {
 		Resilience4JCircuitBreakerFactory factory = new Resilience4JCircuitBreakerFactory(circuitBreakerRegistry,
-				timeLimiterRegistry);
+				timeLimiterRegistry, bulkheadProvider);
 		customizers.forEach(customizer -> customizer.customize(factory));
 		return factory;
 	}
 
+	@Configuration(proxyBeanMethods = false)
+	@ConditionalOnClass(Bulkhead.class)
+	@ConditionalOnProperty(value = "spring.cloud.bulkhead.resilience4j.enable", matchIfMissing = true)
```

Comment:
I think the property should be `spring.cloud.circuitbreaker.bulkhead.resilience4j.enable`

It also needs to be added to this file

https://github.com/spring-cloud/spring-cloud-circuitbreaker/blob/master/spring-cloud-circuitbreaker-resilience4j/src/main/resources/META-INF/additional-spring-configuration-metadata.json

Reply:
enabled is the convention

Reply:
Good call!